cmd/kola/etcd.go

```diff
@@ -48,51 +48,74 @@ func runEtcd(args []string) int {
 		fmt.Fprintf(os.Stderr, "No args accepted\n")
 		return 2
 	}
+
+	var exit int
+	var machines []platform.Machine
+TestLoop:
 	for _, t := range etcdtests.Tests {
+		// kill any machines from the last test instead of waiting for
```

Comment:
How about splitting the body of this loop into a new function. That way you can leave it to a `defer cluster.Destroy()` and clean up a lot of the complexity of this code in general.